### ISSUE TITLE
better to move this to the contructor?

### ISSUE THREAD

**Issue by nirandaperera** [gihankarunarathne/CurwMySQLAdapter#1]
isn't it better to push this 
`m = hashlib.sha256()
`
to the constructor because I think otherwise we will be creating hashlib.sha256() instances everytime we invoke that function, right? :-) 

https://github.com/gihankarunarathne/CurwMySQLAdapter/blob/fde243be642f4ca6b7c84b5a6e822d38e3cb3c09/curwmysqladapter/mysqladapter.py#L74

**Comment by gihankarunarathne** [comment]
Yes, this would be enhance the performance. Will add in further release.

**Comment by gihankarunarathne** [comment]
Seems to be there's an issue with having `hashlib.sha256()` as a single instance for the adapter. May be need to handle to concurrent access.

Need to look into the issue in detail before implementing this feature.
